docs/developer_README.md

```diff
@@ -0,0 +1,55 @@
+# LlamaBot: A Pythonic bot interface to LLMs
+
+This is a `developer` README for LlamaBot.
+
+LlamaBot implements a Pythonic interface to LLMs,
+making it much easier to experiment with LLMs in a Jupyter notebook
+and build Python apps that utilize LLMs.
+All models supported by [LiteLLM](https://github.com/BerriAI/litellm) are supported by LlamaBot.
+
+
+## Setting up your developer environment
+
+Make sure that you have followed the [README.md](../README.md) before jumping on the next steps.
+
+## Creating a Conda virtual environment
+
+It is recommended to have [Conda](https://conda.io/projects/conda/en/latest/user-guide/install/index.html) available on your system.
+
+```bash
+conda env create -f environment.yml
+```
+
+After the conda environment creation, you may need to do run the following commands.
+
+```bash
+conda activate llamabot 
+```
+
+The following command will install all your pip dependencies
+```bash
+pip install -e .
+```
```

Comment:
Could you add in one more section as well please? This is about the pre-commit hooks, which will check for any issues before committing code. I forgot to ask everyone to do it, but within the activated conda environment, one can do `pre-commit install` to install the hooks, so that they get checked locally rather than on CI/CD.

Reply:
Yes, I have added the section now.